lib/enum/basic_enum.rb

```diff
@@ -78,6 +89,7 @@ def groups_of(num)
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_with_object
   # Tip: You'll have to do .each_with_object(Hash.new(0)) to create a hash to start with. Do not save the hash in a variable. You may use the incrementor method here.
   def element_frequency_count
+    collection.each_with_object(Hash.new(0)) { |x, y| y[x] += 1 }
```

Comment:
Definitely a good idea to name your variables better here, because they don't represent what you think. Honestly, each_with_object is pretty weird. Let's try to understand it. For a collection like `['hi', 'hi', 'bye']`, you want it to return the number of times each word appears. So the result should be a new hash like this: `{'hi': 2, 'bye': 1}`

`Hash.new(0)` is probably the weirdest thing here. That expression creates a new hash where every value is 0, and there are no keys (or, ANY key will work). You can try this in pry. 

```
test_hash = Hash.new(0)
test_hash[:asdjfkl] # returns 0
test_hash[1] # returns 0
test_hash[10000] # returns 0
test_hash['hi'] # returns 0
test_hash['bye'] # returns 0
```

So now that we have a hash full of 0's, we can replace some of the 0's with useful values. We can also create some keys for those values.

each_with_object wants two local variables - where you have `|x, y|`. But those variables are NOT `|key, value|`. They are `|element, new_hash|` - the first variable points to the element in the collection, and the second one points to the hash you just created with Hash.new. This is in the Ruby-docs, but I agree it's not obvious!

As you go through the collection one by one, for each element, you create a new key, and then you increment the count by one. 

The full answer looks like this: 

`collection.each_with_object(Hash.new(0)) { |element, new_hash| new_hash[element] += 1 }`